packages/wallet/backend/src/socket/service.ts

```diff
@@ -0,0 +1,58 @@
+import type { Env } from '@/config/env'
+import { Logger } from 'winston'
+import socketIo from 'socket.io'
+import http from 'http'
+
+interface ISocketService {
+  init(args: http.Server): void
+  emitBalanceUpdateByEmail(email: string): void
+}
+
+interface SocketServiceDependencies {
+  logger: Logger
+  env: Env
+}
+
+export class SocketService implements ISocketService {
+  private io: socketIo.Server | null = null
+  private sockets = new Map()
+
+  constructor(private deps: SocketServiceDependencies) {}
+
+  init(server: http.Server) {
+    this.io = new socketIo.Server(server, {
+      cors: {
+        origin: [
+          'http://localhost:4003',
+          `https://${this.deps.env.RAFIKI_MONEY_FRONTEND_HOST}`
+        ],
+        credentials: true
+      }
+    })
+    this.deps.logger.info(`Socket Server is started...`)
+
+    this.io.on('connection', (socket) => {
+      const email = socket.handshake.query.email // Access the user's email
+      this.sockets.set(email, socket)
```

Comment:
Socket.io provides a built-in concept of Rooms. Instead of directly storing sockets in your own data structure, we could use rooms to manage connections. When a user connects, we can add their socket to a room named after their email. This way, we don't need to explicitly manage the sockets ourselves.

```
io.on('connection', (socket) => {
    const email = socket.handshake.query.email
    socket.join(userEmail);
});
```

Reply:
rooms is the way to go, and I suggest to use user id instead of email